src/HttpCache/VarnishPurger.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\HttpCache;
+
+use GuzzleHttp\ClientInterface;
+
+final class VarnishPurger implements PurgerInterface
+{
+    private $clients;
+
+    /**
+     * @param ClientInterface[] $clients
+     */
+    public function __construct(array $clients)
+    {
+        $this->clients = $clients;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function purge(array $iris)
+    {
+        if (!$iris) {
+            return;
+        }
+
+        $parts = array_map(function ($iri) {
+            return sprintf('(^|\,)%s($|\,)', preg_quote($iri));
```

Comment:
You may store the tag with a `,` at the beginning and ending to simplify this RegExp and optimizeinvalidation (especially when you invalidate several tags at the time).
Regexp would became `,%s,`